appservice/src/intTest/groovy/nu/yona/server/UserPhotoTest.groovy

```diff
@@ -0,0 +1,290 @@
+		bobMessagesFromRichard.each
+		{
+			it._links?."yona:userPhoto"?.href == richardPhotoUrl
+		}
+
+		cleanup:
+		appService.deleteUser(richard)
+		appService.deleteUser(bob)
+	}
+
+	def 'Before buddy connect, user photo is already present in buddy request message and is retrievable'()
+	{
+		given:
+		User richard = addRichard()
+		User bob = addBob()
+		def richardPhotoUrl = uploadUserPhoto(richard)
+
+		when:
+		bob.emailAddress = "bob@dunn.net"
+		appService.sendBuddyConnectRequest(richard, bob)
+
+		then:
+		def bobMessages = appService.getMessages(bob)
+		def bobMessagesFromRichard = bobMessages.responseData._embedded?."yona:messages".findAll{ it."nickname" == "RQ"}
+		bobMessagesFromRichard.each
+		{
+			it._links?."yona:userPhoto"?.href == richardPhotoUrl
+		}
+		def response = appService.yonaServer.restClient.get(path: richardPhotoUrl)
+		response.status == 200
+
+		cleanup:
+		appService.deleteUser(richard)
+		appService.deleteUser(bob)
+	}
+
+	def 'After buddy disconnect, user photo is still present in buddy messages and is still retrievable'()
+	{
+		given:
+		def richardAndBob = addRichardAndBobAsBuddies()
+		User richard = richardAndBob.richard
+		User bob = richardAndBob.bob
+		def buddy = appService.getBuddies(richard)[0]
+		def richardPhotoUrl = uploadUserPhoto(richard)
+
+		when:
+		appService.removeBuddy(richard, buddy, "Bob, as you know our ways parted, so I'll remove you as buddy.")
+
+		then:
+		def bobMessages = appService.getMessages(bob)
+		def bobMessagesFromRichard = bobMessages.responseData._embedded?."yona:messages".findAll{ it."nickname" == "RQ"}
+		bobMessagesFromRichard.each
+		{
+			it._links?."yona:userPhoto"?.href == richardPhotoUrl
+		}
+		def response = appService.yonaServer.restClient.get(path: richardPhotoUrl)
+		response.status == 200
+
+		cleanup:
+		appService.deleteUser(richard)
+		appService.deleteUser(bob)
+	}
+
+	def 'After delete of user, user photo is still present in buddy messages and is still retrievable'()
+	{
+		given:
+		def richardAndBob = addRichardAndBobAsBuddies()
+		User richard = richardAndBob.richard
+		User bob = richardAndBob.bob
+		def richardPhotoUrl = uploadUserPhoto(richard)
+
+		when:
+		appService.deleteUser(richard)
+
+		then:
+		def bobMessages = appService.getMessages(bob)
+		def bobMessagesFromRichard = bobMessages.responseData._embedded?."yona:messages".findAll{ it."nickname" == "RQ"}
+		bobMessagesFromRichard.each
+		{
+			it._links?."yona:userPhoto"?.href == richardPhotoUrl
+		}
+		def response = appService.yonaServer.restClient.get(path: richardPhotoUrl)
+		response.status == 200
+
+		cleanup:
+		appService.deleteUser(bob)
+	}
+
+	def 'Buddy updates user photo which causes buddy info change message and user photo update on process'()
+	{
+		given:
+		def richardAndBob = addRichardAndBobAsBuddies()
+		User richard = richardAndBob.richard
+		User bob = richardAndBob.bob
+
+		when:
+		uploadUserPhoto(bob)
```

Comment:
In our test scenarios, Richard is normally the "active user" and Bob the "passive one". Can you swap the two here and in the other tests?

Reply:
Done, is more logical indeed